packages/backend/src/ee/services/ai/types/aiAgent.ts

```diff
@@ -0,0 +1,30 @@
+import { AiChatMessage } from '@lightdash/common';
+import {
+    GetExploreFn,
+    GetPromptFn,
+    RunMiniMetricQueryFn,
+    SearchFieldsFn,
+    SendFileFn,
+    UpdateProgressFn,
+    UpdatePromptFn,
+} from './aiAgentDependencies';
+import { AiAgentExploreSummary } from './aiAgentExploreSummary';
+
+export type AiAgentArgs = {
+    openaiApiKey: string;
+    promptUuid: string;
+    agentName: string;
+    instruction: string | null;
+    messageHistory: AiChatMessage[];
+    aiAgentExploreSummaries: AiAgentExploreSummary[];
+    maxLimit: number;
+};
```

Comment:
Let's iterate on this type, I'd group properties by their purpose:

```suggestion
export type AiAgentArgs = {
    provider: {
      apiKey: string;
      // later stuff like model or baseUrl could go here
    }

    promptUuid: string; // should we go with messageUuid?
    agentName: string;
    instructions: string | null;
    messages: AiChatMessage[];
    exploreSummaries: AiAgentExploreSummary[];
    csvMaxLimit: number; // at first I thought this was `maxSteps`
};
```